truss/remote/baseten/api.py

```diff
@@ -570,6 +570,22 @@ def list_training_jobs(self, project_id: str):
         resp_json = self._rest_api_client.get(f"v1/training_projects/{project_id}/jobs")
         return resp_json
 
+    def list_all_training_jobs(
+        self,
+        status_filter: Optional[List[str]] = None,
+        project_id: Optional[str] = None,
+        job_id: Optional[str] = None,
+    ):
+        resp_json = self._rest_api_client.post(
+            "v1/training_projects/all_jobs",
```

Comment:
What about `POST /v1/training_jobs/search`? I imagine a top level `training_jobs` API is going to be helpful eventually, and I think `search` is more clear than `all_jobs` if I was reading a public API